html/webappapis/dynamic-markup-insertion/opening-the-input-stream/tasks.window.js

```diff
@@ -0,0 +1,114 @@
+// An older version of the HTML Standard mandated that document.open() remove
+// all tasks associated with the document on which open() is called. This step
+// has been proposed to be removed. This series of tests ensures that this step
+// is no longer executed.
+//
+// This file comprehensively (but not exhaustively) tests for many queued tasks
+// that may be observable. Each taskTest() call in fact runs two tests: the
+// first one "tasks without document.open()" does not actually run
+// document.open(), just to test that the tested task works ordinarily; the
+// second actually calls document.open() to test if the method call removes
+// that specific task from the queue.
+
+setup({
+  allow_uncaught_exception: true
+});
+
+function taskTest(description, testBody) {
+  async_test(t => {
+    const frame = document.body.appendChild(document.createElement("iframe"));
+    // The empty HTML seems to be necessary to cajole Chrome into firing a load
+    // event, which is necessary to make sure the frame's document doesn't have
+    // a parser associated with it.
+    frame.src = "/common/blank.html";
+    t.add_cleanup(() => frame.remove());
+    frame.onload = t.step_func(() => {
+      // Make sure there is no parser. Firefox seems to have an additional
+      // non-spec-compliant readiness state "uninitialized", so test for the
+      // two known valid readiness states instead.
+      assert_in_array(frame.contentDocument.readyState, ["interactive", "complete"]);
+      testBody(t, frame, doc => {});
+    });
+  }, `tasks without document.open() (${description})`);
+
+  async_test(t => {
+    const frame = document.body.appendChild(document.createElement("iframe"));
+    // The empty HTML seems to be necessary to cajole Chrome into firing a load
+    // event, which is necessary to make sure the frame's document doesn't have
+    // a parser associated with it.
+    frame.src = "/common/blank.html";
+    t.add_cleanup(() => frame.remove());
+    frame.onload = t.step_func(() => {
+      // Make sure there is no parser. Firefox seems to have an additional
+      // non-spec-compliant readiness state "uninitialized", so test for the
+      // two known valid readiness states instead.
+      assert_in_array(frame.contentDocument.readyState, ["interactive", "complete"]);
+      testBody(t, frame, doc => doc.open());
+    });
+  }, `document.open() and tasks (${description})`);
+}
+
+taskTest("timeout", (t, frame, open) => {
+  let happened = false;
+  // Work around the lint script, since we can't use step_timeout here.
+  frame.contentWindow['setTimeout'](() => happened = true, 100);
```

Comment:
Rather than setting `happened` here, how about just `t.step_func_done()` as the callback to pass the test ASAP?

The following test just times out if there isn't a 2nd message event, so I'd let this test also time out. If that's no good, is there any timeout one could pick that doesn't make the test more likely to be flaky on slow runners?